tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -653,7 +653,7 @@ private List<TopicPartition> filterNoneLeaderPartitions(Collection<TopicPartitio
                 return adminClient.describeTopics(topics).allTopicNames().get().entrySet()
                         .stream()
                         .flatMap(entry -> entry.getValue().partitions().stream()
-                                .filter(partitionInfo -> partitionInfo.leader() == null)
+                                .filter(partitionInfo -> partitionInfo.leader() == null && topicPartitions.contains(new TopicPartition(entry.getKey(), partitionInfo.partition())))
                                 .map(partitionInfo -> new TopicPartition(entry.getKey(), partitionInfo.partition())))
```

Comment:
Please add the filter after `map` to avoid creating extra `TopicPartition` object.
```java
                                .filter(partitionInfo -> partitionInfo.leader() == null)
                                .map(partitionInfo -> new TopicPartition(entry.getKey(), partitionInfo.partition()))
                                .filter(topicPartitions::contains))
```